torchmultimodal/modules/encoders/image_encoder.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Callable, Optional
+
+from torch import nn, Tensor
+from torchmultimodal.modules.layers.transformer import TransformerOutput
+
+
+class VisionTransformer(nn.Module):
+    """
+    General image transformer encoder with embeddings. Similar to ``VisionTransformer`` in torchvision,
+    but more composable. Can be constructed with any user-provided embeddings, encoder, and task head.
+
+    Attributes:
+        embeddings (nn.Module): Module that projects image pixels into embeddings.
+            ``forward()`` should follow interface:
+                images: Optional[Tensor], input data
+                image_patches_mask: Optional[Tensor], mask for patch embeddings
+        encoder (nn.Module): Module for transformer encoder. ``forward()`` should follow interface:
```

Comment:
The same comment as in #249 to leverage cross reference to avoid hard-documenting the interface.